alibi_detect/od/tests/test__lof/test__lof_backend.py

```diff
@@ -0,0 +1,220 @@
+    assert torch.all(outputs.is_outlier == torch.tensor([False, False, True]))
+    assert torch.all(lof_torch(x) == torch.tensor([False, False, True]))
+
+
+def test_lof_torch_backend_ensemble(ensembler):
+    """
+    Test the lof torch backend can be correctly initialized as an ensemble, fit
+    on data and used to predict outliers.
+    """
+
+    lof_torch = LOFTorch(k=[4, 5], ensembler=ensembler)
+    x_ref = torch.randn((1024, 10))
+    lof_torch.fit(x_ref)
+    x = torch.randn((3, 10)) * torch.tensor([[1], [1], [100]])
+    lof_torch.infer_threshold(x_ref, 0.1)
+    outputs = lof_torch.predict(x)
+    assert torch.all(outputs.is_outlier == torch.tensor([False, False, True]))
+    assert torch.all(lof_torch(x) == torch.tensor([False, False, True]))
+
+
+def test_lof_torch_backend_ensemble_ts(tmp_path, ensembler):
+    """
+    Test the lof torch backend can be initialized as an ensemble and
+    torch scripted, as well as saved and loaded to and from disk.
+    """
+
+    lof_torch = LOFTorch(k=[4, 5], ensembler=ensembler)
+    x = torch.randn((3, 10)) * torch.tensor([[1], [1], [100]])
+    x_ref = torch.randn((1024, 10))
+    lof_torch.fit(x_ref)
+    lof_torch.infer_threshold(x_ref, 0.1)
+    pred_1 = lof_torch(x)
+    lof_torch = torch.jit.script(lof_torch)
+    pred_2 = lof_torch(x)
+    assert torch.all(pred_1 == pred_2)
+
+    lof_torch.save(tmp_path / 'lof_torch.pt')
+    lof_torch = torch.load(tmp_path / 'lof_torch.pt')
+    pred_2 = lof_torch(x)
+    assert torch.all(pred_1 == pred_2)
+
+
+def test_lof_torch_backend_ts(tmp_path):
+    """
+    Test the lof torch backend can be initialized and torch scripted, as well as
+    saved and loaded to and from disk.
+    """
+
+    lof_torch = LOFTorch(k=7)
+    x = torch.randn((3, 10)) * torch.tensor([[1], [1], [100]])
+    x_ref = torch.randn((1024, 10))
+    lof_torch.fit(x_ref)
+    lof_torch.infer_threshold(x_ref, 0.1)
+    pred_1 = lof_torch(x)
+    lof_torch = torch.jit.script(lof_torch)
+    pred_2 = lof_torch(x)
+    assert torch.all(pred_1 == pred_2)
+
+    lof_torch.save(tmp_path / 'lof_torch.pt')
+    lof_torch = torch.load(tmp_path / 'lof_torch.pt')
+    pred_2 = lof_torch(x)
+    assert torch.all(pred_1 == pred_2)
+
+
+def test_lof_kernel(ensembler):
+    """
+    Test the lof torch backend can be correctly initialized with a kernel, fit
+    on data and used to predict outliers.
+    """
+
+    kernel = GaussianRBF(sigma=torch.tensor((1)))
+    lof_torch = LOFTorch(k=[4, 5], kernel=kernel, ensembler=ensembler)
+    x_ref = torch.randn((1024, 10))
+    lof_torch.fit(x_ref)
+    x = torch.randn((3, 10)) * torch.tensor([[1], [1], [100]])
+    lof_torch.infer_threshold(x_ref, 0.1)
+    outputs = lof_torch.predict(x)
+    assert torch.all(outputs.is_outlier == torch.tensor([0, 0, 1]))
+    assert torch.all(lof_torch(x) == torch.tensor([0, 0, 1]))
+
+
+@pytest.mark.skip(reason="Can't convert GaussianRBF to torch script due to torch script type constraints")
```

Comment:
What is the intention behind including this test if it is skipped in all cases?

Reply:
I wrote it then decided not to include this functionality in the first set of outliers, however, it should be implemented at some point which is why I've left it in but skipped it. I can take it out if you prefer.